odoo/addons/base/models/ir_module.py

```diff
@@ -307,16 +308,34 @@ def unlink(self):
         self.clear_caches()
         return super(Module, self).unlink()
 
+    @staticmethod
+    def _check_python_external_dependency(pydep):
+        try:
+            pkg_resources.get_distribution(pydep)
+        except pkg_resources.DistributionNotFound as e:
+            try:
+                importlib.import_module(pydep)
+                _logger.warning("python external dependency %s should be replaced by it's PyPI package name", pydep)
+                return  # backward compatibility attempt worked
+            except Exception:
+                pass  # backward compatibility attempt failed
+            _logger.warning("DistributionNotFound: %s", e)
```

Comment:
By indenting the warning and the raise inside the `except` clause above, you will be able to remove the `return` and `pass` lines.

Reply:
@KangOl right, this was a remnant of a version when I wanted to raise the original DistributionNotFound exception. Change applied.